### PR TITLE
feat: Hmac access

### PR DESCRIPTION
**Pull request by vkamlov**
Hi there! 
In this PR I would like to add the possibility of granting HMAC access to service accounts. This will be useful for people, who want to create access to buckets using access keys and secret keys.
If you see what can be better here, feel free to add your changes

### CODE REVIEW

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by vkamlov** [comment]
Changed author and committer info, for passing CLA

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by vkamlov** [comment]
Can someone check this PR? There aren't a lot of changes, only adding a little functionality

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf [hunk 1]
```diff
@@ -220,3 +220,9 @@ resource "google_storage_bucket_object" "folders" {
   name     = "${each.value.folder}/" # Declaring an object with a trailing '/' creates a directory
   content  = "foo"                   # Note that the content string isn't actually used, but is only there since the resource requires it
 }
+
+resource "google_storage_hmac_key" "hmac_keys" {
```
Comment (bharathkkb) [inline main.tf:224]: Lets set project explicitly

File: main.tf [hunk 2]
```diff
@@ -220,3 +220,9 @@ resource "google_storage_bucket_object" "folders" {
...
+  for_each              = var.set_hmac_access ? var.hmac_service_accounts : {}
```
Comment (bharathkkb) [inline main.tf:225]: Can we use var.hmac_service_accounts directly instead of having a bool flag?

**Review by bharathkkb — COMMENTED** [review]
Thanks for the PR @vkamlov and sorry for the delay

File: main.tf (hunk 1 above)
Comment (vkamlov) [reply main.tf:224]: Do you mean to add the next string?
project = var.project_id

File: main.tf (hunk 2 above)
Comment (vkamlov) [reply main.tf:225]: Sorry, I'm not sure that I understand you clearly. Do you want to use it like this?
`for_each = var.hmac_service_accounts`

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

File: main.tf (hunk 1 above)
Comment (bharathkkb) [reply main.tf:224]: Yes, otherwise it will depend on provider default

File: main.tf (hunk 2 above)
Comment (bharathkkb) [reply main.tf:225]: Right since it's {} by default.

File: main.tf (hunk 1 above)
Comment (vkamlov) [reply main.tf:224]: Done

File: main.tf (hunk 2 above)
Comment (vkamlov) [reply main.tf:226]: Without this condition, we can't drop all hmac access using only one variable "set_hmac_access". 
For example, I have a few service accounts with hmac access, and I want to drop them all temporarily. Maybe in the future, I will want to back. And for more convenient usage I can set the variable to false to delete all hmac access without deleting all lists of service accounts from a code.
I think with this condition it is better.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by vkamlov** [comment]
@bharathkkb could you review it again?

**Comment by comment-bot-dev** [comment]
@vkamlov
Thanks for the PR! 🚀<br/>✅ Lint checks have passed.

**Comment by vkamlov** [comment]
Why do the int and lint checks don't execute?

**Comment by vkamlov** [comment]
Thanks! 
Can you merge my changes if everything will be ok?

**Review by apeabody — APPROVED** [review]
Thanks for the contribution @vkamlov!

**Comment by apeabody** [comment]
> Thanks! Can you merge my changes if everything will be ok?

Looks good, done!

**Pull request merged**